Documentation/CircularDependencies.md

```diff
@@ -72,7 +72,7 @@ container.register(ChildProtocol.self) { _ in Child() }
     }
 ```
 
-Here both or either of the depending properties must be specified in the `initCompleted` callback to avoid infinite recursion.
+Here both or either of the depending properties must be specified in the `initCompleted` callback to avoid infinite recursion, however using it on both [has proven to avoid issues](#remark).
```

Comment:
I wouldn't highlight the issues in this section. Put like this it suggests that using `initCompleted` is preferable, which IMO should not be the best practice - we should prefer constructor injection as much as possible. Moreover, in my experience, those issues are usually negligible.

Reply:
this is inside the section "Property injection", from my understanding it is preferred in this scenario, isn't it? This is exactly what I missed the most while reading the docs, once you reach the circular dependencies doc, it is clear that `initCompleted` avoids issues, but until you get there, the docs make you think that it is syntax sugar for setting the property in the registration closure

Reply:
You are right, until reader get's to circular dependencies, `initCompleted` does not have a clear purpose. It has other utilities (it heavily relied on e.g. in `SwinjectStoryboard`), but circular deps is its main purpose...
Maybe we could fix that by adding reference to circular dependencies when showcasing this syntax in previous chapters?